tests/instruments/test_common_base.py

```diff
@@ -255,7 +255,26 @@ def test_remove_child_clears_attributes(self, parent_without_children):
         assert getattr(parent_without_children, "function", None) is None
 
 
-# Test ChildDescriptor
+class TestInheritanceWithChildren:
+    class InstrumentSubclass(Parent):
+        """Override one channel group, inherit other groups."""
+        function = CommonBase.ChannelCreator(GenericBase, "overridden", prefix=None)
+
+    @pytest.fixture()
+    def parent(self):
+        return self.InstrumentSubclass(ProtocolAdapter())
+
+    def test_inherited_children_are_present(self, parent):
+        assert isinstance(parent.ch_A, GenericBase)
+
+    def test_ChannelCreator_is_replaced(self, parent):
```

Comment:
```suggestion
    def test_ChannelCreator_is_replaced_by_channel_collection(self, parent):
```
Maybe? Or add a comment that this is the usual mechanism -- I was initially confused why the changes from `InstrumentSubclass` would _replace_ anything in `.channels`.

Reply:
Good point.